src/screens/OnboardingScreen.tsx

```diff
@@ -1,5 +1,269 @@
+
+  const renderCarouselItem = useCallback(
+    ({ item }: { item: ICarouselItemProps }) => {
+      return (
+        <View
+          style={[
+            styles.carouselItem,
+            // Width of each item should be the full screen - padding
+            { width: width },
+          ]}
+        >
+          <Image style={styles.carouselImage} source={item.image} />
+          <StyledText
+            style={[theme.styles.primaryColor, styles.title]}
+            fontWeight="bold"
+          >
+            {item.title}
+          </StyledText>
+          <StyledText style={[theme.styles.subText, styles.subtext]}>
+            {item.content}
+          </StyledText>
+        </View>
+      );
+    },
+    [width, theme.styles.primaryColor, theme.styles.subText]
+  );
+
+  return (
+    <View style={[theme.styles.screenContainer, styles.container]}>
+      {onExit && (
+        <View style={styles.exitContainer}>
+          <TouchableOpacity onPress={() => onExit()}>
+            <Feather
+              name="x"
+              size={30}
+              color={theme.styles.placeholderText.color}
+            />
+          </TouchableOpacity>
+        </View>
+      )}
+      <FlatList
+        data={CAROUSEL_ITEMS}
+        style={styles.flatList}
+        showsHorizontalScrollIndicator={false}
+        onScroll={Animated.event(
+          [{ nativeEvent: { contentOffset: { x: scrollX } } }],
+          { useNativeDriver: false }
+        )}
+        pagingEnabled
+        onViewableItemsChanged={onViewRef.current}
+        horizontal
+        decelerationRate={"normal"}
+        scrollEventThrottle={16}
+        renderItem={renderCarouselItem}
+      />
+      <View style={styles.footer}>
+        {/* Empty view to make up the left 1/3 of the footer */}
+        <View style={styles.footerItem} />
+        <View style={styles.footerItem}>
+          <ExpandingDot
+            data={CAROUSEL_ITEMS}
+            expandingDotWidth={30}
+            scrollX={scrollX}
+            inActiveDotColor={theme.styles.inactiveColor.color}
+            activeDotColor={theme.styles.primaryColor.color}
+            inActiveDotOpacity={0.5}
+            dotStyle={styles.dotStyles}
+            containerStyle={{}}
+          />
+        </View>
+        <Animated.View
+          style={[
+            styles.footerItem,
+            styles.completeButtonContainer,
+            { opacity: completeFadeButtonAnimation },
+          ]}
+        >
+          {onComplete && (
+            <Button
+              style={styles.completeButton}
+              type="primary"
+              onPress={() => {
+                if (isLastOnboardingPage) {
+                  onComplete();
+                }
+              }}
+            >
+              <Feather name="check" size={20} color="#ffffff" />
+            </Button>
+          )}
+        </Animated.View>
+      </View>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
```

Comment:
We need to look into a better way of managing styles used in components that having this big long list for each of them.

Reply:
I think for our time, its probably for the better

Reply:
Yeah for now it's not a big deal, but maybe down the line it's something we could refactor.